bulk-user-import-sample/BulkUserImport/src/main/java/com/wso2/csa/BulkImportUsers.java

```diff
@@ -0,0 +1,133 @@
+        UserAdminClient userAdminClient = new UserAdminClient(productUrl, session);
+
+        //Uncomment these lines to hardcode the file to be uploaded - For testing purposes
+        //String filePath = "path/to/your/csv/file";
+        //File f = new File(filePath);
+
+        //Uncomment this line to be able to pass the file path as the first parameter of the .jar app
+        File f = new File(props.getFilePath());
+
+        // get the initial time of the process
+        Instant lineCountStart = Instant.now();
+
+        // initiate the variable that count the lines of the .csv file
+        int lines = 0;
+
+        // It will receive the file lines
+        String line = "";
+
+
+        //Iterator that will read the file line by line
+        LineIterator it = FileUtils.lineIterator(f, "UTF-8");
+
+        try {
+            // Taking the useless first line
+            it.nextLine();
+
+            String fileName = f.getName();
+            out.println(fileName);
+
+            // If you want to specify the user store, use the second argument.
+            String userStoreName = props.getUserStoreName();
+
+            // If you want to specify the default password, use the third argument.
+            String defaultPassword = props.getDefaultPassword();
+
+            // Looping through the file lines
+            while (it.hasNext()) {
+
+                //Just counting lines
+                lines++;
+
+                line = it.nextLine();
+
+                //userAdminClient.BulkImportUsers always ignore the first line, that's the \n purpose
+                String newLine = "\n" + line;
+
+                byte[] content = newLine.getBytes();
+                DataHandler file = new DataHandler(new ByteArrayDataSource(content, "application/octet-stream"));
+
+                try {
+                    userAdminClient.BulkImportUsers(userStoreName, defaultPassword, file, fileName);
+                } catch (UserAdminUserAdminException e) {
+                    e.printStackTrace();
+                    out.println("User where error occurred: \n" + line);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            out.println("User where error occurred: \n" + line);
+        } finally {
+            LineIterator.closeQuietly(it);
+        }
+
+        out.println("Total users count: " + lines);
+        Instant lineCountEnd = Instant.now();
+
+        // Calculating the execution time
+        long millis = Duration.between(lineCountStart, lineCountEnd).toMillis();
+        long minutes = (millis / 1000) / 60;
+        millis = millis - 60000 * minutes;
+        long seconds = (millis / 1000) % 60;
+        millis = millis - 1000 * seconds;
+
+
+
+        out.println("Users import time: " +
+                minutes + "m " + seconds + "s " + millis + "ms");
+
+        out.println("Import process finished");
+
+        login.logOut();
+    }
+}
```

Comment:
Let's add a new line at the end of the file.

Reply:
done